razorpay/client.py

```diff
@@ -84,8 +84,20 @@ def _update_user_agent_header(self, options):
     def _get_version(self):
         version = ""
         try: # nosemgrep : gitlab.bandit.B110
-            version = pkg_resources.require("razorpay")[0].version
-        except DistributionNotFound:  # pragma: no cover
+            # Try importlib.metadata first (modern approach)
+            try:
+                import importlib.metadata
+                from importlib.metadata import PackageNotFoundError
+                version = importlib.metadata.version("razorpay")
+            except ImportError:
+                # Fall back to pkg_resources
+                import pkg_resources
+                from pkg_resources import DistributionNotFound
+                version = pkg_resources.require("razorpay")[0].version
+        except (PackageNotFoundError, DistributionNotFound, NameError):  # pragma: no cover
```

Comment:
Catching multiple exceptions ensures robustness, but consider logging a warning if version retrieval fails, to aid client debugging.

Reply:
Catching NameError is thoughtful for edge-case compatibility, but it may also mask unrelated issues. Consider logging or at least adding a comment for future maintainers.